drivers/bluetooth/hci/hci_nxp.c

```diff
@@ -0,0 +1,1178 @@
+			len_to_send = fw_upload_wait_length(true);
+
+			if (len_to_send == V1_START_INDICATION) {
+				continue;
+			}
+			while (len_to_send > 0) {
+				len_to_send = fw_upload_v1_send_data(len_to_send);
+			}
+			if (fw_upload.current_length >= fw_upload.fw_length) {
+				LOG_DBG("FW download done");
+				return 0;
+			}
+			LOG_ERR("FW download failed");
+			return len_to_send;
+		} else if (fw_upload.version == VER3) {
+			if (fw_upload.hdr_sig == V3_START_INDICATION) {
+				fw_upload_wait_req(false);
+				continue;
+			}
+			err = fw_upload_wait_req(false);
+			if (err) {
+				LOG_ERR("Fail to wait req");
+				return err;
+			}
+			if (fw_upload.length) {
+				if (fw_upload.error == 0) {
+					fw_upload_send_ack(V3_REQUEST_ACK);
+					err = fw_upload_v3_send_data();
+					if (err < 0) {
+						LOG_ERR("FW download failed");
+						return err;
+					}
+				} else {
+					LOG_ERR("Error occurs %d", fw_upload.error);
+					fw_upload_send_ack(V3_TIMEOUT_ACK);
+					if (fw_upload.error & BT_MIC_FAIL_BIT) {
+						fw_upload.change_speed_buffer_len = 0;
+						fw_upload.current_length = 0;
+						fw_upload.last_offset = 0;
+					}
+				}
+			} else {
+				if (fw_upload.error == 0) {
+					fw_upload_send_ack(V3_REQUEST_ACK);
+					LOG_DBG("FW download done");
+					return 0;
+				}
+				LOG_ERR("Error occurs %d", fw_upload.error);
+				fw_upload_send_ack(V3_TIMEOUT_ACK);
+				if (fw_upload.error & BT_MIC_FAIL_BIT) {
+					fw_upload.change_speed_buffer_len = 0;
+					fw_upload.current_length = 0;
+					fw_upload.last_offset = 0;
+				}
+			}
+		} else {
+			return -ENOTSUP;
+		}
+	}
+	return -EINVAL;
+}
+
+static void bt_nxp_ctlr_uart_isr(const struct device *unused, void *user_data)
+{
+	int err = 0;
+	int count = 0;
+
+	ARG_UNUSED(unused);
+	ARG_UNUSED(user_data);
+
+	while (uart_irq_update(uart_dev) && uart_irq_is_pending(uart_dev)) {
+		err = uart_poll_in(uart_dev, &fw_upload.rx.buffer[fw_upload.rx.head]);
+		if (err >= 0) {
+			fw_upload.rx.head++;
+			fw_upload.rx.head = fw_upload.rx.head % sizeof(fw_upload.rx.buffer);
+			count++;
+		}
+	}
+
+	while (count > 0) {
+		k_sem_give(&fw_upload.rx.sem);
+		count--;
+	}
+}
+
+int bt_nxp_ctlr_init(void)
+{
+	int err;
+	uint32_t speed;
+	bool flowcontrol_of_hci;
+
+	if (!device_is_ready(uart_dev)) {
+		return -ENODEV;
+	}
+
+	speed = DT_PROP(DT_CHOSEN(zephyr_bt_uart), current_speed);
+	speed = DT_PROP_OR(HCI_UART_CHILD, hci_operation_speed, speed);
+	uart_dev_data.primary_speed = DT_PROP_OR(HCI_UART_CHILD, fw_download_primary_speed, speed);
+	uart_dev_data.secondary_speed =
+		DT_PROP_OR(HCI_UART_CHILD, fw_download_secondary_speed, speed);
+
+	flowcontrol_of_hci = DT_PROP_OR(HCI_UART_CHILD, hw_flow_control, false) ? true : false;
+	uart_dev_data.primary_flowcontrol =
+		DT_PROP_OR(HCI_UART_CHILD, fw_download_primary_flowcontrol, false) ? true : false;
+	uart_dev_data.secondary_flowcontrol =
+		DT_PROP_OR(HCI_UART_CHILD, fw_download_secondary_flowcontrol, false) ? true : false;
```

Comment:
The ternary operator seems redundant here, i.e. `DT_PROP_OR()` should in itself already evaluate to a boolean value.

Reply:
If the property exists, the value is `1`. It needs to be cast to true. Cast it using `bool`.